### ISSUE TITLE
Can't import Youtube subscription

### ISSUE THREAD

**Issue by darrencchang** [TeamNewPipe/NewPipe#1753]
- [x] I carefully read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md) and agree to them.
- [x] I checked if the issue/feature exists in the latest version.
- [ ] I did use the [incredible bugreport to markdown converter](https://teamnewpipe.github.io/CrashReportToMarkdown/) to paste bug reports.

After downloading the xml file from the subscription manager and trying to import it, it displays an error saying "Could not import subscriptions, invalid file/content source

**Comment by sensiblepuffin** [comment]
I tried again just now and it worked. Fairly certain nothing changed between now and when I initially noticed this problem, though.

**Comment by f1vefour** [comment]
I figured it out.

If you try importing your export from:

Subscriptions > Import from > Previous export

It fails.

If you import from:

Subscriptions > Import from > YouTube > IMPORT FILE

It succeeds.

**Comment by theScrabi** [comment]
@mauriciocolli could you please take a look at this?

**Comment by jnihil** [comment]
> I figured it out.
> 
> If you try importing your export from:
> 
> **Subscriptions > Import from > Previous export**
> 
> It fails.
> 
> If you import from:
> 
> **Subscriptions > Import from > YouTube > IMPORT FILE**
> 
> It succeeds.

Yup. That worked for me too.
Thanks for the tip.

**Comment by serussell** [comment]
I exported it from Brave (not the YouTube app) and it worked this way.

**Comment by sulyniv** [comment]
> I wasn't using the app because of this broken feature, but the problem seems to be with downloading it from mobile, somehow.
> 
> What fixed it for me was using the same link they give you https://www.youtube.com/subscription_manager?action_takeout=1
> 
> but opening it on my computer, then putting it on my phone and loading it on the app. It worked this way.

This worked for me, thanks a lot!

**Comment by cuilean** [comment]
> I figured it out.
> 
> If you try importing your export from:
> 
> **Subscriptions > Import from > Previous export**
> 
> It fails.
> 
> If you import from:
> 
> **Subscriptions > Import from > YouTube > IMPORT FILE**
> 
> It succeeds.

thanks bro it worked!

**Comment by Stypox** [comment]
I think this issue should be closed since it was fixed in TeamNewPipe/NewPipeExtractor#160

**Comment by Peskoo** [comment]
> I wasn't using the app because of this broken feature, but the problem seems to be with downloading it from mobile, somehow.
> 
> What fixed it for me was using the same link they give you https://www.youtube.com/subscription_manager?action_takeout=1
> 
> but opening it on my computer, then putting it on my phone and loading it on the app. It worked this way.

Thansk for the tip. It works fine for me.

**Comment by dducro** [comment]
Downloaded the xml file on mac and tried to import it, but it's not working.
Version _0.17.4_

**Comment by Stypox** [comment]
@dducro please open a new issue and provide more information. It would greatly help debugging if you could provide the failing xml file (feel free to truncate personal information) ;-)

**Comment by dducro** [comment]
#2784 Thanks! :)

**Comment by shoukry1** [comment]
Only way I got around it is the following...

**TLDR;**
Export from Google is CSV
Convert into NewPipe export/import style JSON
Export from NewPipe to know requirements and to use JSON skeleton
Convert CSV record into JSON objects and place them within the `subscriptions` array element within the NewPipe JSON export
Then finally import that frankenstein JSON file into NewPipe as "Previous export" instead of "YouTube"

<hr/>

Instructions to export from Google/YouTube are to be slightly revised:
1. Go to the [Google Takeout Manager](https://takeout.google.com/takeout/custom/youtube).
2. Log in if asked.
3. In the "Google Takeout" page, find the "CREATE A NEW EXPORT" section
4. In the "Products" sub-section, Click "Deselect all" to deselect all Google products
5. Scroll all the way down in the Products list to "YouTube"
6. Select "Youtube and Youtube Music" by checking the checkbox right next to it
7. Two sub-options will appear:
  * "Multiple formats"
  * "All YouTube data included"
8. Click on "All YouTube data included"
9. A checklist pop-up will appear "YouTube and YouTube Music content options"
10. Click on "Deselect all" to deselect all content options in the checklist pop-up
11.  Scroll down to find "subscriptions"
12. Only select that option by checking the checkbox right next to it
13. Click "Ok"
14. Click "Next step"
15. In the "Choose file type, frequency & Destination" section, choose the following:
  * Frequency: "Export once"
  * File type & size: ".zip"
15. Click "Create export"
16. Login if needed
17. Wait for the "Download" button to appear and click it
18. Once downloaded, extract the zip file and go down the directory tree as follows:
  * Takeout
  * YouTube and YouTube Music
  * subscriptions
19. You should find a "subscriptions.csv" file (NOT a JSON file as the current instructions expect)
20. Now, for me, I opened that file and tried a few things. Tried using the headers and the data to transform the CSV file into JSON and import it, the app did not like that
21. So I exported my current subscriptions from NewPipe (Doesn't matter if you don't have any yet) (`Menu` -> `Subscriptions` -> `Expand Subscriptions` -> `Export to`: `File`)
22. The NewPipe export is in JSON format and will have the following JSON fields:
  * `app_version`
  * `app_version_int`
  * `subscriptions`
23. The `subscriptions` field will be an array
24. Each JSON object within the `subscriptions` array appears to have the following fields:
  * `service_id`
  * `url`
  * `name`
25. The fields have different names than the CSV headers exported from Google Takeout
26. Seems like the `service_id` field always has a value of `0` in my case (app_version="0.21.9", app_version_int=975)
27. So I used excel to quickly transform the values I exported into a JSON array that looks like what NewPipe expects
28. The CSV export has 3 columns (in Excel, occupied columns **A**, **B** & **C**), with the headers "Channel Id", "Channel Url" & "Channel Title" respectively occupying row 1
29. I placed the following formula in cell **D2**:
  ```
=CONCATENATE("{""service_id"": 0,""url"":""", $B2, """,""name"":""", $C2, """},")
  ```
30. I then copied that formula all the way down till the end of the Excel sheet (CSV opened in Excel)
31. Now that we have a JSON object per row of data, I copied column **D** and pasted it into a text editor
32. I used VSCode, feel free to use your favorite as long as it can format JSON for you and validate it's a proper JSON file before you use it
33. I surrounded my list of JSON objects with the skeleton I exported from NewPipe earlier in _step 21_
34. Then removed the trailing comma (resulting from the formula used in Excel) and made sure the array closure `]` is in place
34. For example:
  ```JSON
  {
    "app_version": "0.21.9",
    "app_version_int": 975,
    "subscriptions": [
        {
            "service_id": 0,
            "url": "http://www.youtube.com/channel/UC-lHJZR3Gqxm24_Vd_AJ5Yw",
            "name": "PewDiePie"
        }
    ]
  }
  ```
35. After formatting the JSON file and validating it has no JSON errors. I then imported it into NewPipe as a "Previous export" instead of "YouTube" (`Menu` -> `Subscriptions` -> `Expand Subscriptions` -> `Import from`: `Previous export`)

Worked like a charm...
Apologies for the very long comment, but I wanted to be thorough for any users who aren't familiar with the app or JSON and whatnot

I'll try to whip-up a quick converter from the Google CSV to the NewPipe JSON if I have some spare time. Should be pretty simple if anyone else wants to give it a go.
This way you won't need to go through _steps 20 - 34_ and it would be automated instead
